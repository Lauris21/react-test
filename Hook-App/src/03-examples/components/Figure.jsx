import { useLayoutEffect, useRef, useState } from "react";

export const Figure = ({ name, sprites, types }) => {
  const h3Ref = useRef();

  const [boxSize, setboxSize] = useState({
    width: 0,
    height: 0,
  });

  useLayoutEffect(() => {
    console.log(h3Ref.current.getBoundingClientRect());
    const { width, height } = h3Ref.current.getBoundingClientRect();

    setboxSize({
      width,
      height,
    });
  }, []);

  return (
    <>
      <figure className="figure text-center" style={{ display: "flex" }}>
        <h3 ref={h3Ref}>{name}</h3>
        <img src={sprites.front_shiny} />
        <div className="container text-center">
          <div className="row align-items-center">
            <p className="col">{types[0].type.name}</p>
            <p className="col">{types[1]?.type.name}</p>
          </div>
        </div>
      </figure>
      <code>{JSON.stringify(boxSize)}</code>
    </>
  );
};
