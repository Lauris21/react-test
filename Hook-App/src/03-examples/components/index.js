export * from "./Figure";
export * from "./IsLoading";
