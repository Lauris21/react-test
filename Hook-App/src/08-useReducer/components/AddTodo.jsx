import { useForm } from "../../hooks/useForm";

export const AddTodo = ({ handleNewTodo }) => {
  const { onInputChange, onResetForm, description } = useForm({
    id: "",
    description: "",
    done: false,
  });

  const onFormSubmit = (e) => {
    e.preventDefault();
    if (description.length < 1) {
      return;
    }

    const newTodo = {
      id: new Date().getTime(),
      description: description,
      done: false,
    };

    handleNewTodo(newTodo);
    onResetForm();
  };

  return (
    <form onSubmit={onFormSubmit}>
      <input
        type="text"
        placeholder="What to do??"
        className="form-control"
        name="description"
        value={description}
        onChange={onInputChange}
      />
      <button type="submit" className="btn btn-outline-primary mt-2">
        ADD
      </button>
    </form>
  );
};
