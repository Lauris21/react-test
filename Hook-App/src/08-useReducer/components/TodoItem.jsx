import PropTypes from "prop-types";

export const TodoItem = ({ item, handleDeleteTodo, onToggleTodo }) => {
  return (
    <li className="list-group-item d-flex justify-content-between">
      <span
        className={`align-self-center ${
          item.done ? "text-decoration-line-through" : ""
        }`}
        aria-label="span"
        onDoubleClick={() => onToggleTodo(item.id)}
      >
        {item.description}
      </span>
      <button
        className="btn btn-danger"
        onClick={() => handleDeleteTodo(item.id)}
      >
        Delete
      </button>
    </li>
  );
};

// TodoItem.proptype = {
//   item: PropTypes.
// }
