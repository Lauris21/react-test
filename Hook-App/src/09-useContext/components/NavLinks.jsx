import { NavLink } from "react-router-dom";

export const NavLinks = ({ text, link }) => {
  return (
    <NavLink
      className={({ isActive }) => `nav-link ${isActive ? "active" : ""}`}
      to={`/${link}`}
    >
      {text}
    </NavLink>
  );
};
