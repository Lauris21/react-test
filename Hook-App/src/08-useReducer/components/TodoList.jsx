import { TodoItem } from "./TodoItem";

export const TodoList = ({ todo, handleDeleteTodo, onToggleTodo }) => {
  return (
    <ul className="list-group p-2">
      {todo.map((item) => (
        <TodoItem
          item={item}
          key={item.id}
          handleDeleteTodo={handleDeleteTodo}
          onToggleTodo={onToggleTodo}
        />
      ))}
    </ul>
  );
};
