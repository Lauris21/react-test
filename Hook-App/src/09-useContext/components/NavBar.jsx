import { Link } from "react-router-dom";
import { NavLinks } from "./NavLinks";

export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-lg bg-body-tertiary rounded-3">
      <div className="container-fluid">
        <Link className="navbar-brand" to="/">
          UseContext
        </Link>
        <div className="collapse navbar-collapse" id="navbarNav">
          <ul className="navbar-nav">
            <li className="nav-item">
              <NavLinks text={"Home"} link={""} />
            </li>
            <li className="nav-item">
              <NavLinks text={"About"} link={"about"} />
            </li>
            <li className="nav-item">
              <NavLinks text={"Login"} link={"login"} />
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};
