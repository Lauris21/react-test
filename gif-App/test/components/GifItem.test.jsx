import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Prueba para <GifItem />", () => {
  const title = "Probando Título";
  const url = "http://probando_url.jpg/";
  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar la imagen con el url y el alt correctos", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole("img").src).toBe(url)

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("Debe mostrar el título en el componente", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
