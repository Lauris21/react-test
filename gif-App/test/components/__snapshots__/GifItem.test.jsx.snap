// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba para <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Probando Título"
      src="http://probando_url.jpg/"
    />
    <p>
      Probando Título
    </p>
  </div>
</div>
`;
